x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_list/components/collapsed_item_actions.tsx

```diff
@@ -71,15 +78,19 @@ export const CollapsedItemActions: React.FunctionComponent<ComponentOpts> = ({
             name="disable"
             disabled={!item.isEditable || !item.enabledInLicense}
             compressed
-            checked={!item.enabled}
+            checked={isDisabled}
             data-test-subj="disableSwitch"
             onChange={async () => {
-              if (item.enabled) {
-                await disableAlert(item);
-              } else {
-                await enableAlert(item);
-              }
-              onAlertChanged();
+              const enabled = !isDisabled;
+              asyncScheduler.schedule(async () => {
```

Comment:
Verified that this works but curious if we need to use `asyncScheduler` here? On the alert details page, we optimistically set `isEnabled` and `isMuted` when an action is taken. Can we do something similar here?

```
const enabled = !isDisabled;
if (enabled) {
    setIsDisabled(true);
    await disableAlert({ ...item, enabled });
} else {
    setIsDisabled(false);
    await enableAlert({ ...item, enabled });
}
onAlertChanged();
```

Would that be enough here as well? Or does `asyncScheduler` give us something else?

Reply:
If we clicked multiple times on the switch control we are expecting that the calls of enable/disable will be done in the same order. So that is why I've added this asyncScheduler